pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java

```diff
@@ -96,6 +96,8 @@ public enum TransformFunctionType {
   LOOKUP("lookUp"),
   GROOVY("groovy"),
 
+  EXTRACT_YEAR("extractYear"),
```

Comment:
We want to add one `EXTRACT` transform function for all time unit, instead of one function for each unit

Reply:
Yes, with the help of only `EXTRACT`, the overall logic became much simpler.